### DISCUSSION TITLE
Cannot install behind a proxy with emacs26.1 and spacemacs0.300

### DISCUSSION THREAD

**Discussion by TradDog** [syl20bnr/spacemacs#15101] [Q&A]
#### Description :octocat:
When installing behind a corporate proxy, packages are not found and error message, "package name mispelled ?" is issued.

#### Reproduction guide :beetle:
Behind a proxy:
- emacs 26.1 pre-installed on Rhel8 machine. 
- git clone https://github.com/syl20bnr/spacemacs ~/.emacs.d -> ok. 
- start emacs: answering setup questions -> ok.
-  download start: package not found. 

Then:
followed https://github.com/syl20bnr/spacemacs/issues/4807
instructions. In particular, inserted 
```
(setq url-proxy-services
   '(("no_proxy" . "^\\(localhost\\|10.*\\)")
     ("http" . "your.proxy.com.cn:80")
     ("https" . "your.proxy.com.cn:80")))

(setq url-http-proxy-basic-auth-storage
    (list (list "your.proxy.com.cn:80"
                (cons "Input your LDAP UID !"
                      (base64-encode-string "your-user-name:your-password")))))

```
Being asked many times credentials. 

*Observed behaviour:* :eyes: :broken_heart:
Failed to download most packages. 

*Expected behaviour:* :heart: :smile:
full installation of spacemacs

(%LAST_KEYS%)
OS: Linux Rhel8
Emacs: 26.1
Spacemacs: 0.300

#### Backtrace :paw_prints:
Corporate environment prohibits trace and logs to be exported. 

In essence, could instructions on handling proxy with spacemacs 0.300 been 
restated ?

**Comment by lebensterben** [comment]
proxy setup is not spacemacs related. transferring to discussion.

**Comment by c02y** [comment]
I tried because of the similar reason, didn't make it work inside init.el.

Using proxychains in terminal to launch emacs to install/update packages.

Since you are Spacemacs, once I want to install/update packages, I use:
```
emacs --batch -l ~/.emacs.d/init.el --eval="(configuration-layer/update-packages t)"
```

Setup your procychains.conf file, and 
```
proxychains emacs --batch -l ~/.emacs.d/init.el --eval="(configuration-layer/update-packages t)"
```
